### PR TITLE
MacOs build fix proposal.

### PR DESCRIPTION
**Pull request by devnexen**
Made on Big Sur, removing few unnecessary old specifics.
Build with LLVM homebrew.

### CODE REVIEW

**Comment by devnexen** [comment]
is able to build and start 
image

**Comment by xddxd** [comment]
Doesn't seem to work on AppleClang? 
`rpcs3/util/atomic.hpp:1003:20: error: couldn't allocate output register for constraint 'x'`

**Comment by devnexen** [comment]
I built with `WITH_LLVM=OFF` and LLVM/clang homebrew

**Comment by Nekotekina** [comment]
Can you try to build with LLVM on?

File: rpcs3/Emu/NP/fb_helpers.cpp [hunk 1]
```diff
@@ -195,7 +195,7 @@ u16 np_handler::RoomDataInternal_to_SceNpMatching2RoomDataInternal(const RoomDat
 			{
 				if (room_info->roomGroup[g_index].groupId == member->roomGroup())
 				{
-					member_info->roomGroup = vm::cast(room_info->roomGroup.addr() + (sizeof(SceNpMatching2RoomGroup) * g_index));
+					member_info->roomGroup = vm::cast(room_info->roomGroup.addr() + (static_cast<u32>(sizeof(SceNpMatching2RoomGroup)) * g_index));
```
Comment (Nekotekina) [inline rpcs3/Emu/NP/fb_helpers.cpp:198]:
```suggestion
					member_info->roomGroup = vm::cast(room_info->roomGroup.addr() + (u32{sizeof(SceNpMatching2RoomGroup)} * g_index));
```

**Comment by devnexen** [comment]
> Can you try to build with LLVM on?

It does not work with.

**Comment by Nekotekina** [comment]
Hmm okay, we can revisit it later.

**Comment by xddxd** [comment]
It's possible to compile a build with LLVM, you need to comment out one else if block. It compiles modules and everything, but when it runs for example the triangle test, it segfaults.

**Comment by devnexen** [comment]
> It's possible to compile a build with LLVM, you need to comment out one else if block. It compiles modules and everything, but when it runs for example the triangle test, it segfaults.

Probably possible but then maybe that is why it segfaults :-) anyway I may have a look at the whole thing this weekend unless someone find first.

**Pull request merged**